freeplane/src/main/java/org/freeplane/view/swing/map/MiniMapView.java

```diff
@@ -0,0 +1,343 @@
+package org.freeplane.view.swing.map;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.Cursor;
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Insets;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.RenderingHints;
+import java.awt.Toolkit;
+import java.awt.event.ComponentAdapter;
+import java.awt.event.ComponentEvent;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseWheelEvent;
+import java.awt.geom.AffineTransform;
+import java.awt.image.BufferedImage;
+
+import javax.swing.BorderFactory;
+import javax.swing.BoundedRangeModel;
+import javax.swing.ImageIcon;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JRootPane;
+import javax.swing.JScrollBar;
+import javax.swing.JScrollPane;
+import javax.swing.JViewport;
+import javax.swing.SwingUtilities;
+import javax.swing.event.MouseInputAdapter;
+
+import org.freeplane.api.LengthUnit;
+import org.freeplane.core.resources.IFreeplanePropertyListener;
+import org.freeplane.core.resources.ResourceController;
+import org.freeplane.features.map.IMapChangeListener;
+import org.freeplane.features.map.MapChangeEvent;
+import org.freeplane.features.mode.Controller;
+import org.freeplane.features.styles.MapStyle;
+import org.freeplane.features.ui.FrameController;
+import org.freeplane.features.ui.IMapViewManager;
+import org.freeplane.features.ui.ViewController;
+
+public class MiniMapView extends JPanel implements IFreeplanePropertyListener, IMapChangeListener {
```

Comment:
I don't know for sure.

I searched for possible mini-map implementations in swing and the existing component hierarchy
came to me first, so I tried for several iterations and finally this PR.

It's one of the typical looks of mini-map, while it's also very attractive to use resizable and arbitrary positioned dialog. I'm acceptable with this existing implementation because the mini-map can be easily switched on/off (that's the reason I added the control-M mapping at the beginning).

I don't know how the public users would choose.

Reply:
I agree with you.
Let me know when you finish your work and I can merge it.